packages/OreIdReact/src/components/Observables/Observables.spec.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+	render as renderTest,
+	RenderOptions,
+	waitFor,
+} from "@testing-library/react";
+import React from "react";
+import { createTestOreId, createTestWebWidget } from "src/test-utils";
+import { Observables } from "./Observables";
+import { OreIdContext } from "src/OreIdContext";
+import { OreIdWebWidget } from "oreid-webwidget";
+import { OreId } from "oreid-js";
+
+interface Options extends RenderOptions {
+	providerProps: {
+		oreId: OreId;
+		webWidget: OreIdWebWidget;
+		user: any;
+		setUser: (user: any) => void;
+		isLoggedIn: boolean;
+		setIsLoggedIn: (isLoggedIn: boolean) => void;
+		accessToken: string | undefined;
+		setAccessToken: (accessToken: string | undefined) => void;
+	};
+}
+
+const render = (
+	ui: JSX.Element,
+	{ providerProps, ...renderOptions }: Options
+) => {
+	return renderTest(
+		<OreIdContext.Provider
+			value={{
+				...providerProps,
+			}}
+		>
+			{ui}
+		</OreIdContext.Provider>,
+		renderOptions
+	);
+};
+
+test("Should render children", async () => {
```

Comment:
I'm not familiar with patterns in testing UI components. Is it good practice to include tests in the component like this?

Reply:
Ah nvm ... I see this is a spec file.